.github/workflows/npm-publish.yml

```diff
@@ -23,13 +23,20 @@ jobs:
   publish-npm:
     needs: build
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 20
-          registry-url: https://registry.npmjs.org/
+          node-version: 22.x
+          registry-url: "https://registry.npmjs.org"
+          cache: "npm"
       - run: npm ci
-      - run: npm publish
+      - run: npm run build --if-present
+      - run: npm version ${TAG_NAME} --git-tag-version=false
```

Comment:
Does this mean we can create releases without having to push a version bump to main? 

Will we lose the tags and version in our package file by doing this? Sounds nice being able to release without the extra push but also I like having tags and the right version in the JSON 😅

Reply:
> Does this mean we can create releases without having to push a version bump to main?

Yes.

> Will we lose the tags and version in our package file by doing this? Sounds nice being able to release without the extra push but also I like having tags and the right version in the JSON 😅

Also yes. Typically I set package.json version to `0.0.0-development` so it's clear that the version is generated.

Reply:
No worries I think that makes sense. @koddsson can you update the package version so that is clear?